src/core/reopt_inputs.jl

```diff
@@ -485,44 +492,28 @@ function setup_tech_inputs(s::AbstractScenario, time_steps)
     seg_min_size, seg_max_size, seg_yint, techs_by_exportbin, export_bins_by_tech, boiler_efficiency,
     tech_renewable_energy_fraction, tech_emissions_factors_CO2, tech_emissions_factors_NOx, tech_emissions_factors_SO2, 
     tech_emissions_factors_PM25, techs_operating_reserve_req_fraction, thermal_cop, fuel_cost_per_kwh, 
-    heating_cop, cooling_cop, heating_cf, cooling_cf, avoided_capex_by_ashp_present_value
+    heating_cop, cooling_cop, heating_cf, cooling_cf, avoided_capex_by_ashp_present_value,
+    pbi_pwf, pbi_max_benefit, pbi_max_kw, pbi_benefit_per_kwh
 end
 
 
 """
-    setup_pbi_inputs(s::AbstractScenario, techs::Techs)
+    setup_pbi_inputs!(techs::Techs, tech::AbstractTech, tech_name::String, financial::Financial,
+        pbi_pwf, pbi_max_benefit, pbi_max_kw, pbi_benefit_per_kwh
+    )
 
-Create data arrays for production based incentives. 
-All arrays can be empty if no techs have production_incentive_per_kwh > 0.
+Modifies dictionaries for production based incentives: pbi_pwf, pbi_max_benefit, pbi_max_kw, pbi_benefit_per_kwh. 
+All dicts can be empty if no techs have production_incentive_per_kwh > 0.
 """
-function setup_pbi_inputs(s::AbstractScenario, techs::Techs)
-
-    pbi_pwf = Dict{String, Any}()
-    pbi_max_benefit = Dict{String, Any}()
-    pbi_max_kw = Dict{String, Any}()
-    pbi_benefit_per_kwh = Dict{String, Any}()
-
-    for tech in techs.all
```

Comment:
tech in techs.all were all strings, so the code below was not working as intended.